### ISSUE TITLE
Extend CI testing with more "smoke tests"

### ISSUE THREAD

**Issue by remtav** [NRCan/geo-deep-learning#242]
Inspired by: https://mattsegal.dev/devops-academic-research.html

Until we write unit tests for critical modules, our CI testing should include at least 2 or three different usage scenarios as "smoke test" (see [Smoke Test section](https://mattsegal.dev/devops-academic-research.html)) to ensure nothing breaks.

Some scenarios to test:
- Use [visualization](https://github.com/NRCan/geo-deep-learning/blob/develop/utils/visualization.py) features
- Use [deeplabv3 with concatenation point for NIR band](https://github.com/NRCan/geo-deep-learning/blob/develop/models/model_choice.py#L203)
- Use debug mode
- Use at least one more loss (ex.: sgd)
- Use at least one model architecture from segmentation models pytorch (ex.: [Unet pretrained](https://github.com/NRCan/geo-deep-learning/blob/develop/models/model_choice.py#L28))
- etc.

**Comment by ymoisan** [comment]
Have a look at [https://docs.docker.com/ci-cd/best-practices/](https://docs.docker.com/ci-cd/best-practices/)  :how you can use tests in your Docker project and re-use them in the CI.
